src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorShape.cs

```diff
@@ -473,7 +472,7 @@ public nint AdjustToNextIndex(in TensorShape destinationShape, nint linearOffset
 
                     if (index < length)
                     {
-                        break;
+                        return linearOffset;
```

Comment:
We've talked about this logic a few times. A small comment covering why returning linearOffset here vs resetting back to `-strides[^1]` as the final would be beneficial

Reply:
Added, let me know your thoughts.

Reply:
This isn't quite the comment I was thinking of, I think we need one giving a simple example of like `src = length: [...]; stride: [...]; dst = length: [...]; stride: [...]` (or similar) which explains why such iteration logic is then correct.

In general the first loop handles every index that exists in `src` while anything in the second loop handles additional indexes in `dst`. The indexes in `dst` will only flip over in the case that the most significant index in `src` rolls over, at which point every index from `src` will be `0` and we should be repeating the data from `src`.

So it's still not clear when returning `linearOffset` would be correct, since we should be at an index that only exists in the destination due to a broadcast.